src/markdown-pages/docs/managers/core-manager.md

```diff
@@ -3,12 +3,247 @@ path: "/docs/managers/core-manager"
+| Attribute: `infixes` | Set selection precedence on the property that you want to extract. See [Code snippets and examples](#codesnippets) for more information. Default is {}|
+| Attribute: `required` |  Default is `true` |
+| Syntax:  |<code>@TestProperty(prefix = "string", suffix = "string", infixes = "string", required = true)</code> |
+
+</details>
+
+
+
+# <a name="codesnippets"></a>Code snippets and examples
+
+<details><summary>Extract credentials from the Galasa credentials store</summary>
+
+You can extract credentials by using the `getCredentials` method. The Core Manager uses the `getCredentials` method to retrieve a user id and password from the credentials store to use in your test.
+
+```
+import dev.galasa.ICredentials;
+import dev.galasa.ICredentialsUsernamePassword;
+...
+@Test
+...
+ICredentialsUsernamePassword credentials = (ICredentialsUsernamePassword) \
+coreManger.getCredentials("SIMBANK");
+credentials.getPassword();
+credentials.getUsername();
+```
+
+You can edit the credentials.properties file in your .galasa folder. The following example shows the contents of the credentials.properties file that is set up as part of Galasa SIMBANK tutorials:
+
+```
+secure.credentials.SIMBANK.username=IBMUSER
+secure.credentials.SIMBANK.password=SYS1
+```
+
+</details>
+
+<details><summary>Mask the value of an extracted password</summary>
+
+To mask the password, for example to prevent it from being displayed in recorded screens, use the Core Manager `registerConfidentialText` method.
+
+```
+coreManager.registerConfidentialText("SYS1", "IBMUSER password");
+```
+</details>
+
+<details><summary>Store an XML request in the test results archive</summary>
+
+Use the following example code to understand how to archive messages in a particular folder and file structure in the RAS:
+
+```
+Path requestPath = artifactRoot.resolve(communications).resolve(messages).;
+Files.write(requestPath, content.getBytes(), new SetContentType(ResultArchiveStoreContentType.TEXT),
+StandardOpenOption.CREATE);
+```
+
+Messages in the RAS are stored in the following structure:
+
+```
+communications
+|
+---messages 
+```
+
+If the folder or file do not exist, it is created by using the `resolve` method.
+
+</details>
+
+
+<details><summary>Using the Test Property annotation</summary>
+
+This example has the following CPS properties set in the CPS file: 
+
+test.projectA.first.choice.data=3
+test.projectA.choice.data=2
+
+where `projectA` is the prefix, `data` is the suffix, and `first` and `choice` are infixes.
+
+The following code is used in the test:
+
+```
+@TestProperty(prefix = "projectA", suffix = "data", infixes = {"first","choice"})
+public String property;
+```
+
+In this example, if the property _test.projectA.first.choice.data_ is found in the CPS, then this is extracted for use in the test. If _test.projectA.first.choice.data_ is not found, then property _test.projectA.choice.data_ is used instead.
+
```

Comment:
In this example, if the property _test.projectA.first.choice.data_ is found in the CPS, then this is extracted for use in the test. If _test.projectA.first.choice.data_ is not found, then property _test.projectA.first.data_ is used instead.

Reply:
updated thank you